File: scratch/check_vanshika.py
import os
import json
from supabase import create_client, Client
from dotenv import load_dotenv

load_dotenv(".env.local")
url = os.environ.get("NEXT_PUBLIC_SUPABASE_URL")
key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")
supabase: Client = create_client(url, key)

# Get Vanshika's user_id
user_res = supabase.table('users').select('*').eq('username', 'zerodata502_Vanshika').execute()
if not user_res.data:
    print("User Vanshika not found.")
else:
    user_id = user_res.data[0]['user_id']
    print(f"Vanshika user_id: {user_id}")
    
    # Get total calls
    calls = supabase.table('call_logs').select('*').eq('user_id', user_id).execute()
    print(f"Total calls for Vanshika: {len(calls.data)}")
    
    # Analyze outcomes
    outcomes = {}
    for c in calls.data:
        out = c.get('outcome', 'None')
        outcomes[out] = outcomes.get(out, 0) + 1
        
    print("Outcomes:", outcomes)
    
    # Are there any followups?
    followups = [c for c in calls.data if c.get('next_followup_date')]
    print(f"Total followups: {len(followups)}")

File: scratch/check_vanshika.js
import { createClient } from '@supabase/supabase-js';
import * as dotenv from 'dotenv';
dotenv.config({ path: '.env.local' });

const supabase = createClient(process.env.NEXT_PUBLIC_SUPABASE_URL, process.env.SUPABASE_SERVICE_ROLE_KEY);

async function main() {
    const { data: users, error: userErr } = await supabase.from('users').select('*').eq('username', 'zerodata502_Vanshika');
    if (userErr) {
        console.error(userErr);
        return;
    }
    if (!users || users.length === 0) {
        console.log("User Vanshika not found.");
        return;
    }
    const userId = users[0].user_id;
    console.log("Vanshika user_id:", userId);
    
    const { data: calls, error: callsErr } = await supabase.from('call_logs').select('*').eq('user_id', userId);
    if (callsErr) {
        console.error(callsErr);
        return;
    }
    console.log(`Total calls for Vanshika: ${calls.length}`);
    
    const outcomes = {};
    const followups = [];
    calls.forEach(c => {
        const out = c.outcome || 'None';
        outcomes[out] = (outcomes[out] || 0) + 1;
        if (c.next_followup_date) {
            followups.push(c);
        }
    });
    console.log("Outcomes:", outcomes);
    console.log(`Total followups: ${followups.length}`);
}

main().catch(console.error);
